container/sagemaker-tensorflow-inferentia/build_artifacts/sagemaker/tfs_utils.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import logging
+import multiprocessing
+import os
+import re
+
+from collections import namedtuple
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+DEFAULT_CONTENT_TYPE = "application/json"
+DEFAULT_ACCEPT_HEADER = "application/json"
+CUSTOM_ATTRIBUTES_HEADER = "X-Amzn-SageMaker-Custom-Attributes"
+
+Context = namedtuple("Context",
+                     "model_name, model_version, method, rest_uri, grpc_port, "
+                     "custom_attributes, request_content_type, accept_header, content_length")
+
+
+def parse_request(req, rest_port, grpc_port, default_model_name, model_name=None):
+    tfs_attributes = parse_tfs_custom_attributes(req)
+    tfs_uri = make_tfs_uri(rest_port, tfs_attributes, default_model_name, model_name)
+
+    if not model_name:
+        model_name = tfs_attributes.get("tfs-model-name")
+
+    context = Context(model_name,
+                      tfs_attributes.get("tfs-model-version"),
+                      tfs_attributes.get("tfs-method"),
+                      tfs_uri,
+                      grpc_port,
+                      req.get_header(CUSTOM_ATTRIBUTES_HEADER),
+                      req.get_header("Content-Type") or DEFAULT_CONTENT_TYPE,
+                      req.get_header("Accept") or DEFAULT_ACCEPT_HEADER,
+                      req.content_length)
```

Comment:
Minor: since it's a namedtuple, should use the keyword args.

Reply:
Copied from SM TF container.